kobo-uncaged/device/device.go

```diff
@@ -580,41 +565,39 @@ func (k *Kobo) SaveDeviceInfo() error {
 }
 
 // SaveCoverImage generates cover image and thumbnails, and save to appropriate locations
-func (k *Kobo) SaveCoverImage(contentID string, size image.Point, imgB64 string) {
-	defer k.Wg.Done()
-
+func (k *Kobo) SaveCoverImage(contentID string, size image.Point, imgB64 string, done chan<- struct{}) {
 	img, _, err := image.Decode(base64.NewDecoder(base64.StdEncoding, strings.NewReader(imgB64)))
 	if err != nil {
 		log.Println(err)
+		done <- struct{}{}
```

Comment:
You could `defer` this instead.

Reply:
Of course I could. Thanks for catching that.